chromium_src/chrome/browser/browser_features.cc

```diff
@@ -13,6 +13,7 @@ OVERRIDE_FEATURE_DEFAULT_STATES({{
 #if !defined(OS_ANDROID)
     {kCopyLinkToText, base::FEATURE_DISABLED_BY_DEFAULT},
 #endif
+    {kDestroyProfileOnBrowserClose, base::FEATURE_DISABLED_BY_DEFAULT},
```

Comment:
do we want to have different behavior than Chrome here? @diracdeltas @pes10k ?

Reply:
This is effectively private browsing mode, no?

Reply:
From https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/flag_descriptions.cc;l=1692:
```
    "Release memory and other resources when a Profile's last browser window "
    "is closed, rather than when Chrome closes completely.";
```
Also from https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/browser_features.cc;l=28:
```
// Destroy profiles when their last browser window is closed, instead of when
// the browser exits.
```
So it doesn't seem related to private browsing per-se, but more like not waiting until the browser binary exits to destroy the profiles used by its windows, as that could happen way after the last window is closed (e.g. on Windows, if you have Chrome running in the background, it will be running even if no windows are open, see the icon in the taskbar).

Cc @emerick in any case

Reply:
Right, this feature destroys the profile as soon as the browser closes vs. waiting until the browser exits. With this feature enabled, `ClearOnExit()`'s call to `GetLoadedProfiles()` returns no profiles because they've already been destroyed earlier, so `ClearOnExit` ends up deleting nothing.

It seems like we might want to tie clearing the data to the profile lifetime, but I wasn't too sure what the scope of that change might be. I can enter an issue for that work. It doesn't seem like it needs to hold up this PR, but let me know what you think.

Reply:
@emerick what doesn't work when this is enabled? If the call to `GetLoadedProfiles` returns nothing, doesn't that mean we've already destroyed them? cc @yrliou

Reply:
@bridiver Yeah, `GetLoadedProfiles` returns no profiles. It does mean that we've destroyed the profiles, but my understanding is that "clear data on exit" is currently tied to browser exit and not profile destruction.

Reply:
BTW, just to clarify, the feature name is referring to destruction of the profile in the C++ sense via a destructor (not deleting the profile from disk).

Reply:
@yrliou @darkdh shouldn't we be setting `kForceEphemeralProfiles` on tor/session profiles so they automatically get deleted at the correct time? Or call `ScheduleProfileForDeletion` when the profile is created?
```
// Schedules the profile at the given path to be deleted on shutdown. If we're
  // deleting the last profile, a new one will be created in its place, and in
  // that case the callback will be called when profile creation is complete.
  void ScheduleProfileForDeletion(const base::FilePath& profile_dir,
```